SCloudRandomizer/MainViewController.m

```diff
@@ -234,23 +234,34 @@ - (void)enableButtons:(BOOL)enable {
     [self.btnInfo setEnabled:enable];
     [self.btnLike setEnabled:enable];
     [self.btnChangeParams setEnabled:enable];
+    [self.btnSCDisconnect setEnabled:enable];
+    [self.lblArtistValue setEnabled:enable];
+    [self.lblTitleValue setEnabled:enable];
+    [self.lblCurrentTime setEnabled:enable];
+    [self.lblLengthValue setEnabled:enable];
+    [self.scrubber setEnabled:enable];
     
     if (enable) {
         self.imgArtwork.alpha = 1;
     } else {
-        self.imgArtwork.alpha = 0.5;
-        self.imgArtwork.layer.borderWidth = 1;
+        self.imgArtwork.alpha = 0.3;
     }
 }
 
 - (void)playTrack {
-    if (self.isPlayingForFirstTime) {
+    if (self.isPlayingForFirstTime && self.prepToPlayHud == nil) {
         self.prepToPlayHud = [self getProgressBar:MBProgressHUDModeIndeterminate progressHudLabel:@"Preparing to play" progressHudDetailsLabel:@"Please wait.."];
```

Comment:
If it is playing for the first time, shouldn't self.prepToPlayHud be null? If its not, then perhaps we are hiding some other problem>